src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -148,5 +177,235 @@ private void InitializeBuilder()
                     .WithObserverFactory(this);
             }
         }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await GetMetricsAsync();
+        }
+
+        internal async Task<CosmosDBTriggerMetrics> GetMetricsAsync(IRemainingWorkEstimator workEstimator)
+        {
+            _workEstimator = workEstimator;
+            return await GetMetricsAsync();
+        }
+
+        public async Task<CosmosDBTriggerMetrics> GetMetricsAsync()
+        {
+            int partitionCount = 0;
+            long remainingWork = 0;
+            IReadOnlyList<RemainingPartitionWork> workPerPartition = null;
+
+            try
+            {
+                InitializeBuilder();
+
+                if (_workEstimator == null)
+                {
+                    _workEstimator = await this._hostBuilder.BuildEstimatorAsync();
+                }
+
+                workPerPartition = await this._workEstimator.GetEstimatedRemainingWorkPerPartitionAsync();
+
+                if (workPerPartition != null)
+                {
+                    partitionCount = workPerPartition.Count;
+                    remainingWork = workPerPartition.Sum(item => item.RemainingWork);
+                }
+            }
+            catch (Microsoft.Azure.Documents.DocumentClientException e)
+            {
+                if (!TryHandleDocumentClientException(e))
+                {
+                    _logger.LogWarning("Unable to handle Microsoft.Azure.Documents.DocumentClientException: {0}", e.Message);
+                }
+            }
+            catch (NullReferenceException e)
+            {
+                // This shouldn't be hit because we're not using ChangeFeedHost, and WorkEstimator throws DocumentClientException
+                // for missing collections.
+                _logger.LogWarning(e.Message);
+            }
+            catch (InvalidOperationException e)
```

Comment:
Not sure about all this error handling. Are you trying to identify transient errors? I don't like parsing the error strings looking for things (KnownDocumentClientErrors). Thinking we should be able to simplify and just log any exceptions we receive as Warnings + failure to get scale status. @ealsur may have an opinion as well

Reply:
`DocumentClientException` would mean something happened at runtime trying to access the leases, for example, the user deleted the collections, or, we got throttled. `InvalidOperationException` would mean there is some issue with the input parameters, some parameter is null or missing.

I guess a nullref can happen if the user configuration is empty or it's pointing at a connection string that does not exist?

Reply:
@mathewc those are valid points. These catch blocks, as well as the code related to KnownDocumentClientErrors, were all pulled from the existing scaling logic in ScaleController. So the code here is primarily my attempt to match the existing logging as closely as possible so we emit similar errors and troubleshooting info. I can definitely clean it up and do it the way you describe if that's what we would prefer.

Reply:
@ealsur might have been the one who wrote the original ScaleController code? If so and if he's OK with this then I guess it's fine. The whole idea of parsing these error messages is very fragile and if Cosmos starts returning different error messages, we'll silently start missing them.  Ideally the Cosmos errors would be structured and return an error ID we could key off of. Anyhow let's not block on this for the first rev.

Reply:
I didn't write the ScaleController code, so I'm not very familiar with this logic, but I agree with your other comment regarding unifying them if they are similar.

Reply:
After talking with @mathewc I think we decided to leave this error handling as it is for now (matching ScaleController), but I've removed the null ref block so we don't just swallow them.